test/test-ctrl-c-timeout.js

```
@@ -0,0 +1,211 @@
+
+    // Verify state was reset
+    assert.strictEqual(stdinManager.ctrlCCount, 0, 'Ctrl+C count should be reset to 0');
+    assert.strictEqual(stdinManager.ctrlCTimer, null, 'Timer should be null');
+
+    // Verify ANSI clear sequences were output
+    assert(output.includes('\x1b[1A'), 'Output should contain cursor up sequence');
+    assert(output.includes('\r'), 'Output should contain carriage return');
+    assert(output.includes('\x1b[K'), 'Output should contain clear line sequence');
+
+    // Count occurrences
+    const upCount = (output.match(/\x1b\[1A/g) || []).length;
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)